FlightJournal.Web/Controllers/FlightController.cs

```diff
@@ -77,9 +78,24 @@ public ViewResult Grid(DateTime? date, int? locationid)
 
             var flights = this.db.Flights.Where(s => (!locationid.HasValue || (s.LandedOn.LocationId == locationid.Value || s.StartedFrom.LocationId == locationid.Value)) && (date.HasValue ? s.Date == date : s.Date == DateTime.Today))
                 .Include("Plane").Include("StartedFrom").Include("LandedOn").Include("Pilot").Include("PilotBackseat").Include("Betaler").Include("StartType")
-                .OrderByDescending(s => s.Date).ThenByDescending(s => s.Departure ?? DateTime.Now);
-
-            return View(flights.ToList().Where(f => f.IsCurrent()));
+                .OrderByDescending(s => s.Date).ThenByDescending(s => s.Departure ?? DateTime.Now)
+                .ToList()
+                .Where(f=>f.IsCurrent())
+                .ToList();
+
+            var flyingToday = flights.Select(f => f.PilotId).ToList();
+
+            var last12months = DateTime.Now.AddYears(-1);
```

Comment:
1) **klub relation** ? skal vi vise for alle piloter eller for piloter som har samme klub som den person der kigger på grid. 
2) **Performance hit** som også drøftet er til aften, hvis vi kan tillade os at sige der er tale om status fra start på dagen, så kan vi tillade os at låse data til fra i går og huske det resten af dagen på hver pilot som opstår i listen. Mistænker også at trace på sql kaldet hvor der er "flyingToday.Contains" måske kan optimeres.

Reply:
re 1: Det er vist egentlig allerede klaret da flyingToday kun indeholder piloter fra det aktuelt viste grid ?
re 2: jeg kunne egentlig godt tænke mig at vi ser hvad performance er før vi evt forsøger at optimere. Der er jo et relativt lille antal piloter på et Grid, og hvis Azure (?) kan lave bare nogenlunde fornuftige queries, så burde det ikke være dyrt. Den med trace forstod jeg ikke ?

Reply:
1. ok
2. ok
2b. Når du afvikler i debug og åbner en trace så får du den SQL der bliver kaldt til database med direkte i trace view. Jeg kunne mistænke at det kald hvor flyingToday indgår måske laver en grim SQL